Common/Deathray.cs

```diff
@@ -18,71 +18,71 @@ protected Deathray(float MaxTime, float LaserLength, float rotation, Color laser
             this.MaxTime = MaxTime;
             this.laserColor = laserColor;
             this.LaserLength = LaserLength;
-            this.projectile.ai[0] = rotation;
+            this.Projectile.ai[0] = rotation;
             this.TexturePath = texture;
         }
         public float RotationalSpeed
         {
-            get => projectile.ai[0];
-            set => projectile.ai[0] = value;
+            get => Projectile.ai[0];
+            set => Projectile.ai[0] = value;
         }
         public override void SetDefaults()
         {
-            projectile.width = 10;
-            projectile.height = 10;
-            projectile.penetrate = -1;
-            projectile.tileCollide = false;
+            Projectile.width = 10;
+            Projectile.height = 10;
+            Projectile.penetrate = -1;
+            Projectile.tileCollide = false;
         }
         public string TexturePath;
         public Color laserColor = Color.White; // just color
-        public Texture2D LaserBeginTexture => mod.GetTexture(TexturePath);//Main.projectileTexture[projectile.type];
-        public Texture2D LaserMiddleTexture => mod.GetTexture(TexturePath + "1");
-        public Texture2D LaserEndTexture => mod.GetTexture(TexturePath + "2");
+        public Texture2D LaserBeginTexture => ModContent.Request<Texture2D>(TexturePath).Value;//Main.projectileTexture[Projectile.type];
+        public Texture2D LaserMiddleTexture => ModContent.Request<Texture2D>(TexturePath + "1").Value;
+        public Texture2D LaserEndTexture => ModContent.Request<Texture2D>(TexturePath + "2").Value;
         public float LaserLength = 1000; // laser lenght
-        public float timer { get { return projectile.localAI[0]; } set { projectile.localAI[0] = value; } }
+        public float timer { get { return Projectile.localAI[0]; } set { Projectile.localAI[0] = value; } }
         public float MaxTime = 0; // time the laser lasts
-        public void DrawLaser(SpriteBatch spriteBatch, Color beamColor, float scale)
+        public void DrawLaser(Color beamColor, float scale)
         {
-            spriteBatch.Draw(LaserBeginTexture, projectile.Center - Main.screenPosition, null, laserColor * projectile.Opacity, projectile.rotation, new Vector2(LaserBeginTexture.Width * scale, LaserBeginTexture.Height) / 2, new Vector2(scale, 1f), SpriteEffects.None, 0f);
+            Main.EntitySpriteDraw(LaserBeginTexture, Projectile.Center - Main.screenPosition, null, laserColor * Projectile.Opacity, Projectile.rotation, new Vector2(LaserBeginTexture.Width * scale, LaserBeginTexture.Height) / 2, new Vector2(scale, 1f), SpriteEffects.None, 0);
```

Comment:
There's a _lot_ of this. In particular, there is no longer a `spriteBatch` being sent into PreDraw/PostDraw for projectiles and items (NPCs still get them). My understanding is that it's to do custom filtering and stuff whenever you draw. It uses Main.spriteBatch if you need to modify blend state.